Warning about potential value truncation

```
src/h5cpp/node/dataset.cpp:243:73: warning: absolute value function 'abs' given an argument of type 'ssize_t' (aka 'long') but has parameter of type 'int' which may cause truncation of value [-Wabsolute-value]
if((delta<0) && (current_dims[dimension_index] < static_cast<hsize_t>(abs(delta))))
                                                                         ^
src/h5cpp/node/dataset.cpp:243:73: note: use function 'std::abs' instead
   if((delta<0) && (current_dims[dimension_index] < static_cast<hsize_t>(abs(delta))))
                                                                         ^~~
                                                                         std::abs
```

`::abs` uses `int`, `::std::abs` is defined for `int`,`long int`, and `long long int`.

--- a/src/h5cpp/node/dataset.cpp
+++ b/src/h5cpp/node/dataset.cpp
@@ -26,6 +26,7 @@
 // Created on: Sep 12, 2017
 //
 
+#include <cstdlib>
 #include <sstream>
 #include <stdexcept>
 #include <h5cpp/node/dataset.hpp>
@@ -240,7 +241,7 @@ void resize_by(const Dataset &dataset,size_t dimension_index,ssize_t delta)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wsign-compare"
 #endif
-  if((delta<0) && (current_dims[dimension_index] < static_cast<hsize_t>(abs(delta))))
+  if((delta<0) && (current_dims[dimension_index] < static_cast<hsize_t>(std::abs(delta))))
   {
     std::stringstream ss;
     ss<<"Extent of dataset ["<<dataset.link().path()<<"] cannot be changed "

